### PR TITLE
Conversion PearDatabase to yii [IT529]

### PR DESCRIPTION
**Pull request by areksolek**
Conversion PearDatabase to yii [IT529]

### CODE REVIEW

File: vtlib/Vtiger/PackageExport.php [hunk 1]
```diff
@@ -278,25 +278,17 @@ public function exportDependencies(ModuleBasic $moduleInstance)
...
 		$moduleId = $this->moduleInstance->id;
-
-		$sqlresult = $adb->pquery('SELECT * FROM vtiger_tab WHERE tabid = ?', [$moduleId]);
-		$tabInfo = $adb->getRow($sqlresult);
-
-		$tabname = $tabInfo['name'];
-		$tablabel = $tabInfo['tablabel'];
-		$tabVersion = $tabInfo['version'] ?? false;
-
+		$dataQuery = (new \App\Db\Query())->select(['name', 'tablabel', 'version'])->from('vtiger_tab')->where(['tabid' => $moduleId])->createCommand()->queryOne();
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/PackageExport.php]:
```suggestion
		$row = (new \App\Db\Query())->select(['name', 'tablabel', 'version'])->from('vtiger_tab')->where(['tabid' => $moduleId])->one();
```

File: vtlib/Vtiger/PackageExport.php [hunk 2]
```diff
@@ -424,85 +399,75 @@ public function exportBlocks(ModuleBasic $moduleInstance)
...
 				$this->closeNode('picklistvalues');
 			}
-
 			// Export field to module relations
-			if ($uitype == '10') {
-				$relatedmodres = $adb->pquery('SELECT * FROM vtiger_fieldmodulerel WHERE fieldid=?', [$fieldid]);
-				$relatedmodcount = $adb->numRows($relatedmodres);
-				if ($relatedmodcount) {
+			if ('10' == $uiType) {
+				$dataReader = (new \App\Db\Query())->from('vtiger_fieldmodulerel')->where(['fieldid' => $row['fieldid']])->createCommand()->query();
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/PackageExport.php]: column()

File: vtlib/Vtiger/PackageExport.php [hunk 3]
```diff
@@ -514,32 +479,36 @@ public function exportFields(ModuleBasic $moduleInstance, $blockid)
...
 				$this->outputNode('', 'fieldparams');
 				$this->openNode('tree_template');
-				$trees = $adb->pquery('SELECT * FROM vtiger_trees_templates WHERE templateid=?;', [$fieldresultrow['fieldparams']]);
-				if ($adb->numRows($trees) > 0) {
-					$this->outputNode($adb->queryResultRaw($trees, 0, 'name'), 'name');
-					$this->outputNode($adb->queryResultRaw($trees, 0, 'access'), 'access');
-					$treesData = $adb->pquery('SELECT * FROM vtiger_trees_templates_data WHERE templateid=?;', [$fieldresultrow['fieldparams']]);
+				$treesExist = (new \App\Db\Query())->select(['name', 'access'])->from('vtiger_trees_templates')->where(['templateid' => $fieldParams]);
+				$treesValue = $treesExist->all();
+				if ($treesExist->exists()) {
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/PackageExport.php]:
```suggestion
				if ($treesValue) {
```

File: vtlib/Vtiger/PackageExport.php [hunk 4]
```diff
@@ -514,32 +479,36 @@ public function exportFields(ModuleBasic $moduleInstance, $blockid)
...
+			if ('302' == $uiType) {
 				$this->outputNode('', 'fieldparams');
 				$this->openNode('tree_template');
-				$trees = $adb->pquery('SELECT * FROM vtiger_trees_templates WHERE templateid=?;', [$fieldresultrow['fieldparams']]);
-				if ($adb->numRows($trees) > 0) {
-					$this->outputNode($adb->queryResultRaw($trees, 0, 'name'), 'name');
-					$this->outputNode($adb->queryResultRaw($trees, 0, 'access'), 'access');
-					$treesData = $adb->pquery('SELECT * FROM vtiger_trees_templates_data WHERE templateid=?;', [$fieldresultrow['fieldparams']]);
+				$treesExist = (new \App\Db\Query())->select(['name', 'access'])->from('vtiger_trees_templates')->where(['templateid' => $fieldParams]);
+				$treesValue = $treesExist->all();
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/PackageExport.php]:
```suggestion
				$treesValue = $treesExist->one();
```

File: vtlib/Vtiger/PackageExport.php [hunk 5]
```diff
@@ -424,85 +399,75 @@ public function exportBlocks(ModuleBasic $moduleInstance)
...
-
-		if (empty($fieldcount)) {
+		$dataReader = (new \App\Db\Query())->from('vtiger_field')->where(['tabid' => $moduleInstance->id, 'block' => $blockid])->createCommand()->query();
+		if (0 === $dataReader->count()) {
 			return;
 		}
-
-		$entityresult = $adb->pquery('SELECT * FROM vtiger_entityname WHERE tabid=?', [$moduleInstance->id]);
-		$entity_fieldname = $adb->queryResult($entityresult, 0, 'fieldname');
+		$entityField = (new \App\Db\Query())->select(['fieldname', 'entityidfield', 'entityidcolumn'])->from('vtiger_entityname')->where(['tabid' => $moduleInstance->id])->all();
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/PackageExport.php]:
```suggestion
		$entityField = (new \App\Db\Query())->select(['fieldname', 'entityidfield', 'entityidcolumn'])->from('vtiger_entityname')->where(['tabid' => $moduleInstance->id])->one();
```

File: vtlib/Vtiger/PackageExport.php [hunk 6]
```diff
@@ -597,33 +565,30 @@ public function exportCustomViews(ModuleBasic $moduleInstance)
...
 	public function exportSharingAccess(ModuleBasic $moduleInstance)
 	{
-		$adb = \PearDatabase::getInstance();
-
-		$deforgshare = $adb->pquery('SELECT * FROM vtiger_def_org_share WHERE tabid=?', [$moduleInstance->id]);
-		$deforgshareCount = $adb->numRows($deforgshare);
-
-		if (empty($deforgshareCount)) {
+		$permission = (new \App\Db\Query())->select(['permission'])->from('vtiger_def_org_share')->where(['tabid' => $moduleInstance->id])
+			->column();
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/PackageExport.php]: ?

File: vtlib/Vtiger/PackageExport.php [hunk 7]
```diff
@@ -640,19 +605,16 @@ public function exportActions(ModuleBasic $moduleInstance)
 		if (!$moduleInstance->isentitytype) {
 			return;
 		}
-
-		$adb = \PearDatabase::getInstance();
-		$result = $adb->pquery('SELECT distinct(actionname) FROM vtiger_profile2utility, vtiger_actionmapping
-			WHERE vtiger_profile2utility.activityid=vtiger_actionmapping.actionid and tabid=?', [$moduleInstance->id]);
-
-		if ($adb->numRows($result)) {
+		$dataReader = (new \App\Db\Query())->select(['actionname'])->from('vtiger_profile2utility')->innerJoin('vtiger_actionmapping', 'vtiger_profile2utility.activityid=vtiger_actionmapping.actionid')->where(['tabid' => $moduleInstance->id])->distinct('actionname')->createCommand()->query();
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/PackageExport.php]: vtiger_profile2utility.activityid = vtiger_actionmapping.actionid

File: vtlib/Vtiger/PackageExport.php [hunk 8]
```diff
@@ -248,26 +248,26 @@ public function __copyLanguageFiles(\App\Zip $zip, $module)
...
-		for ($i = 0; $i < $noOfPreferences; ++$i) {
-			$prefName = $adb->queryResult($sqlResult, $i, 'prefname');
-			$prefValue = $adb->queryResult($sqlResult, $i, 'prefvalue');
-			if ($prefName == 'vtiger_min_version') {
-				$minVersion = $prefValue;
-			}
-			if ($prefName == 'vtiger_max_version') {
-				$maxVersion = $prefValue;
+		$dataReader = (new \App\Db\Query())->from('vtiger_tab_info')->where(['tabid' => $moduleId])->createCommand()->query();
+		if (0 < $dataReader->count()) {
```
Comment (rskrzypczak) [inline vtlib/Vtiger/PackageExport.php]: remove this 'if'

File: vtlib/Vtiger/PackageExport.php [hunk 9]
```diff
@@ -514,32 +478,35 @@ public function exportFields(ModuleBasic $moduleInstance, $blockid)
...
-					for ($i = 0; $i < $countTreesData; ++$i) {
-						$this->openNode('tree_value');
-						$this->outputNode($adb->queryResultRaw($treesData, $i, 'name'), 'name');
-						$this->outputNode($adb->queryResultRaw($treesData, $i, 'tree'), 'tree');
-						$this->outputNode($adb->queryResultRaw($treesData, $i, 'parentTree'), 'parentTree');
-						$this->outputNode($adb->queryResultRaw($treesData, $i, 'depth'), 'depth');
-						$this->outputNode($adb->queryResultRaw($treesData, $i, 'label'), 'label');
-						$this->outputNode($adb->queryResultRaw($treesData, $i, 'state'), 'state');
-						$this->closeNode('tree_value');
+					if (0 < $dataReaderRow->count()) {
```
Comment (rskrzypczak) [inline vtlib/Vtiger/PackageExport.php]: remove this 'if'

File: vtlib/Vtiger/PackageExport.php [hunk 10]
```diff
@@ -667,67 +628,55 @@ public function exportRelatedLists(ModuleBasic $moduleInstance)
...
 					$this->openNode('actions');
-					$actions = explode(',', $action_text);
+					$actions = explode(',', $actionText);
 					foreach ($actions as $action) {
 						$this->outputNode($action, 'action');
 					}
 					$this->closeNode('actions');
 				}
-				$this->closeNode('inrelatedlist');
+				$this->openNode('inrelatedlist');
```
Comment (rskrzypczak) [inline vtlib/Vtiger/PackageExport.php]:
```suggestion
				$this->closeNode('inrelatedlist');
```

**Pull request merged**